packages/components/src/components/icon/icon.tsx

```diff
@@ -58,13 +63,7 @@ const Icon = React.forwardRef(
                 onMouseEnter={onMouseEnter}
                 onMouseLeave={onMouseLeave}
                 ref={ref}
-                style={
-                    custom_color
-                        ? {
-                              '--fill-color1': custom_color,
-                          }
-                        : undefined
```

Comment:
@jim-deriv why do we need this variable here outside? 🙂
```
: {}
```

Reply:
@maryia-binary, The reason I extracted the expression wasn't due to `undefined`. It's to do with the need to give the `{
                '--fill-color1': custom_color,
            }` object a type. I could use the `{}: type` syntax here, but this wasn't possible due to the use of the ternary operator.

Reply:
@jim-deriv you're right, TS doesn't recognize our custom '--fill-color1' as a CSS property. So your solution is fine, although you can also consider this one (working for me)  🙂 :
```
style={
                    (custom_color
                        ? {
                              '--fill-color1': custom_color,
                          }
                        : undefined) as CSSProperties & { '--fill-color1': string }
                }
```